src/legacy/ui/public/courier/fetch/components/shard_failure_description_header.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { EuiCode } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { ShardFailure } from './shard_failure_types';
+import { formatKey } from './shard_failure_description';
+
+export function getFailurePropsForSummary(
+  failure: ShardFailure
+): Array<{ key: string; value: string }> {
+  const failureDetailProps: Array<keyof ShardFailure> = ['shard', 'index', 'node'];
+  return failureDetailProps
+    .filter(key => typeof failure[key] === 'number' || typeof failure[key] === 'string')
+    .map(key => ({ key, value: String(failure[key]) }));
+}
+
+export function getFailureSummaryText(failure: ShardFailure, failureDetails?: string): string {
+  const failureName = formatKey(failure.reason.type);
+  const displayDetails =
+    typeof failureDetails === 'string' ? failureDetails : getFailureSummaryDetailsText(failure);
+
+  return i18n.translate('common.ui.courier.fetch.shardsFailedModal.failureHeader', {
+    defaultMessage: '{failureName} at {failureDetails}',
+    values: { failureName, failureDetails: displayDetails },
+    description: 'Summary of a shard failure',
```

Comment:
From my experience with questions coming from translators, it would be beneficial if we add an example to this description, e.g.: 

`Summary of shard failures, e.g. "IllegalArgumentException at shard 0 node xyz"`

Reply:
true! done